salt/states/pkgrepo.py

```diff
@@ -702,6 +708,12 @@ def absent(name, **kwargs):
         ret["comment"] = "Failed to configure repo '{}': {}".format(name, exc)
         return ret
 
+    if repo and "pkg.apt_source_entry" in __salt__:
+        if set(__salt__["pkg.apt_source_entry"](stripname)["architectures"]) != set(
```

Comment:
Note: I have not spoke to the team yet, but will today. I just had a follow up thought after yesterday. Could this logic be added to the `get_repo` function instead? Right here specifically https://github.com/saltstack/salt/blob/master/salt/modules/aptpkg.py#L2008  we could check if arch exists and compare the two.

Reply:
No, I tried this to begin with and doing so breaks `pkgrepo.managed`.

Reply:
K, thanks for confirming.